### PR TITLE
[plug-in] enable running of VS Code extension tests

### PR DESCRIPTION
**Pull request by akosyakov**
fix https://github.com/theia-ide/theia/issues/4226

Changes needed to run VS Code API unit tests: https://github.com/theia-
ide/theia/issues/4224#issuecomment-459752557

### CODE REVIEW

File: packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts [hunk 1]
```diff
@@ -117,6 +118,9 @@ export class HostedPluginProcess implements ServerPluginRunner {
         env.PATH = process.env.PATH;
         // add HOME to env since some plug-ins need to read files from user's home dir
         env.HOME = process.env.HOME;
+        if (argv.extensionTestsPath) {
```
Comment (benoitf) [inline packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts]: it's not overkill to use bring yargs dependency there ?

Comment (akosyakov) [reply packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts]: `yargs` should be always available

but I better to replace it with `CliContribution` that we parse args only once

File: packages/plugin-ext/src/plugin/plugin-manager.ts [hunk 2]
```diff
@@ -120,10 +122,13 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
             if (pluginMain !== undefined) {
                 this.startPlugin(plugin, configStorage, pluginMain);
             } else {
-                return Promise.reject(new Error('Unable to load the given plugin'));
+                console.error(`Unable to load a plugin from "${plugin.pluginPath}"`);
```
[the remark this answers is not in the document]
Comment (akosyakov) [reply packages/plugin-ext/src/plugin/plugin-manager.ts]: The issue here that VS Code extension contributes only to package.json and stub `main` property with an inexistent file or just skip it. It causes an exception and prevents loading other plugins and tests.

[the remark this answers is not in the document]
Comment (benoitf) [reply packages/plugin-ext/src/plugin/plugin-manager.ts]: ok that's not really nice to specify `main` to a file that does not exist :-/

**Comment by akosyakov** [comment]
@benoitf i've addressed your comments, could you have a look again please?

**Pull request merged**